logical/logical.go

```diff
@@ -2,6 +2,18 @@ package logical
 
 import log "github.com/mgutz/logxi/v1"
 
+// BackendType is the type of backend that is being implemented
+type BackendType string
+
+// The these are the types of backends that can be derived from
+// logical.Backend
+const (
+	TypeUnknown     BackendType = "unknown"
+	TypeLogical     BackendType = "logical"
```

Comment:
Are these string values ever shown to users? If not, can they be made uint vals? If so, let's rename them to what users expect ("secret", "auth" for instance)